components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/EndpointRegistryImpl.java

```diff
@@ -234,4 +236,23 @@ public void updateEndpointRegistry(String registryId, String registryName, Endpo
 
     }
 
+    /**
+     * Creates a new version of an Endpoint Registry Entry
+     *
+     * @param entryId       Registry Entry Identifier(UUID)
+     * @param registryEntry EndpointRegistryEntry
+     * @param version       New version of the Registry Entry
+     * @return entryID UUID of the created Registry Entry
+     * @throws APIManagementException if failed to delete the Endpoint Registry Entry
+     */
+    public String createNewEntryVersion(String entryId, EndpointRegistryEntry registryEntry, String version)
+            throws APIManagementException {
+        registryEntry.setVersion(version);
```

Comment:
Can't we set the version to ```EndpointRegistryEntry``` object  in API impl class?

Reply:
Fixed in 87488e4